### DISCUSSION TITLE
Fetch all users having "permission x" when using teams

### DISCUSSION THREAD

**Discussion by fractalf** [spatie/laravel-permission#2186] [Q&A]
Hi,
I'm struggling to see how one would fetch all users having a certain permission when using **teams**

Ref. https://spatie.be/docs/laravel-permission/v5/basic-usage/basic-usage
```php
$users = User::permission('edit articles')->get(); // Returns only users with the permission 'edit articles' (inherited or directly)
```

This doesn't return anything, ..unless I use the `setPermissionsTeamId`
```php
$teamId = 1;
setPermissionsTeamId($teamId);
$users = User::permission('edit articles')->get(); // Returns only users with the permission 'edit articles' (inherited or directly)
```

Fetching all the teams and then do this in a for loop is pretty bad, so how would you suggest doing this?

**UPDATE**

I managed to do this *manually*, but one would think there was a better way..

```php
$permission = Permission::where('name', 'edit articles')->first();

$res = DB::table('role_has_permissions')
    ->select('role_id')
    ->where('permission_id', $permission->id)
    ->get();

$roleIds = $res->map(function($item) {
    return $item->role_id;
})->toArray();

$res = DB::table('model_has_roles')
    ->select('model_id')
    ->where('model_type', \App\Models\User::class)
    ->whereIn('role_id', $roleIds)
    ->get();

$userIds = $res->map(function($item) {
    return $item->model_id;
})->toArray();

$users = User::whereIn('id', $userIds)->get();

```

**Comment by erikn69** [comment]
Better make your own `scope` and your own `relation` for that

**Comment by fractalf** [comment]
Oookay, this whole team thing seems very "patched", but I guess it's not easy to incorporate this idea into an existing lib..

**Reply by parallels999** [reply]
Feel free to make a PR so that each time it can be better for everyone
